### PR TITLE
add measure result

### PR DESCRIPTION
**Pull request by qibobo**

### CODE REVIEW

File: pkg/command/service/measure.go [hunk 1]
```diff
@@ -151,11 +158,11 @@ kperf service measure --svc-perfix svc --range 1,200 --namespace ns --concurrenc
 						}
 						if !svcIns.IsReady() {
 							fmt.Printf("service %s/%s not ready and skip measuring\n", svc, svcNs)
-							measureArgs.notReadyCount = measureArgs.notReadyCount + 1
+							currentMeasureResult.notReadyCount = currentMeasureResult.notReadyCount + 1
```
Comment (maximilien) [inline pkg/command/service/measure.go:161]: I think you can simplify this with `currentMeasureResult.notReadyCount += 1`

Comment (maximilien) [reply pkg/command/service/measure.go:161]: Same for other statements below...

File: pkg/command/service/measure.go [hunk 2]
```diff
@@ -326,35 +333,55 @@ kperf service measure --svc-perfix svc --range 1,200 --namespace ns --concurrenc
...
-						measureArgs.svcReadySum = measureArgs.svcReadySum + svcReadyDuration.Seconds()
-						svcReadyTime = append(svcReadyTime, svcReadyDuration.Seconds())
+						currentMeasureResult.svcConfigurationsReadySum = currentMeasureResult.svcConfigurationsReadySum + svcConfigurationsReadyDuration.Seconds()
+						currentMeasureResult.revisionReadySum = currentMeasureResult.revisionReadySum + revisionReadyDuration.Seconds()
+						currentMeasureResult.deploymentCreatedSum = currentMeasureResult.deploymentCreatedSum + deploymentCreatedDuration.Seconds()
+						currentMeasureResult.podScheduledSum = currentMeasureResult.podScheduledSum + podScheduledDuration.Seconds()
+						currentMeasureResult.containersReadySum = currentMeasureResult.containersReadySum + containersReadyDuration.Seconds()
+						currentMeasureResult.queueProxyStartedSum = currentMeasureResult.queueProxyStartedSum + queueProxyStartedDuration.Seconds()
+						currentMeasureResult.userContrainerStartedSum = currentMeasureResult.userContrainerStartedSum + userContrainerStartedDuration.Seconds()
```
Comment (maximilien) [inline pkg/command/service/measure.go:342]: Same as above to simplify and make it easier to read...

**Review by maximilien — CHANGES_REQUESTED** [review]
Missing tests. Left a few suggestions to simplify

**Comment by qibobo** [comment]
@maximilien 
Thank you for your comments. @gyohuangxin is working on the ut. I have fixed other comments.

File: pkg/command/service/measure.go [hunk 3]
```diff
@@ -326,35 +331,56 @@ kperf service measure --svc-perfix svc --range 1,200 --namespace ns --concurrenc
...
-
 			if len(svcNamespacedName) == 0 {
 				return errors.New("no service found to measure")
 			}
 
 			for _, item := range svcNamespacedName {
-				svcChannel <- item
 				group.Add(1)
+				svcChannel <- item
+
```
Comment (zhanggbj) [inline pkg/command/service/measure.go:360]: maybe remove extra new line

**Review by zhanggbj — APPROVED** [review]

File: pkg/command/service/common.go [hunk 4]
```diff
@@ -52,7 +52,9 @@ type measureArgs struct {
 	namespaceRange  string
 	namespacePrefix string
 	concurrency     int
-
+	verbose         bool
+}
+type measureResult struct {
 	svcConfigurationsReadySum   float64
 	svcRoutesReadyReadySum      float64
```
Comment (zyjiaobj) [inline pkg/command/service/common.go:59]: Why we have 2 `Ready` in the param name?

Comment (zyjiaobj) [reply pkg/command/service/common.go:59]: As well for `ingressReadyReadySum `.

File: pkg/command/service/measure.go [hunk 5]
```diff
@@ -326,35 +331,56 @@ kperf service measure --svc-perfix svc --range 1,200 --namespace ns --concurrenc
...
+				measureFinalResult.revisionReadySum += workerMeasureResults[i].revisionReadySum
+				measureFinalResult.deploymentCreatedSum += workerMeasureResults[i].deploymentCreatedSum
+				measureFinalResult.podScheduledSum += workerMeasureResults[i].podScheduledSum
+				measureFinalResult.containersReadySum += workerMeasureResults[i].containersReadySum
+				measureFinalResult.queueProxyStartedSum += workerMeasureResults[i].queueProxyStartedSum
+				measureFinalResult.userContrainerStartedSum += workerMeasureResults[i].userContrainerStartedSum
+				measureFinalResult.svcRoutesReadyReadySum += workerMeasureResults[i].svcRoutesReadyReadySum
+				measureFinalResult.ingressReadyReadySum += workerMeasureResults[i].ingressReadyReadySum
+				measureFinalResult.ingressNetworkConfiguredSum += workerMeasureResults[i].ingressNetworkConfiguredSum
+				measureFinalResult.ingressLoadBalancerReadySum += workerMeasureResults[i].ingressNetworkConfiguredSum
```
Comment (zyjiaobj) [inline pkg/command/service/measure.go:376]: Shouldn't it be `workerMeasureResults[i].ingressLoadBalancerReadySum`

File: pkg/command/service/measure.go [hunk 6]
```diff
@@ -377,82 +403,82 @@ kperf service measure --svc-perfix svc --range 1,200 --namespace ns --concurrenc
...
 				fmt.Printf("- Service Revision Duration:\n")
-				fmt.Printf("  Total: %fs\n", float64(measureArgs.revisionReadySum))
-				fmt.Printf("  Average: %fs\n", float64(measureArgs.revisionReadySum)/float64(measureArgs.readyCount))
+				fmt.Printf("  Total: %fs\n", float64(measureFinalResult.revisionReadySum))
+				fmt.Printf("  Average: %fs\n", float64(measureFinalResult.revisionReadySum)/float64(measureFinalResult.readyCount))
 
 				fmt.Printf("  - Service Deployment Created Duration:\n")
-				fmt.Printf("    Total: %fs\n", float64(measureArgs.revisionReadySum))
-				fmt.Printf("    Average: %fs\n", float64(measureArgs.revisionReadySum)/float64(measureArgs.readyCount))
+				fmt.Printf("    Total: %fs\n", float64(measureFinalResult.revisionReadySum))
```
Comment (zyjiaobj) [inline pkg/command/service/measure.go:419]: Shouldn't it be `measureFinalResult.deploymentCreatedSum`?

File: pkg/command/service/measure.go [hunk 7]
```diff
@@ -377,82 +403,82 @@ kperf service measure --svc-perfix svc --range 1,200 --namespace ns --concurrenc
...
+				fmt.Printf("    Average: %fs\n", float64(measureFinalResult.revisionReadySum)/float64(measureFinalResult.readyCount))
```
Comment (zyjiaobj) [inline pkg/command/service/measure.go:420]: Shouldn't it be measureFinalResult.deploymentCreatedSum

**Review by zyjiaobj — APPROVED** [review]
LGTM

**Comment by zhanggbj** [comment]
/lgtm
/assign @maximilien 
/assign @zyjiaobj 

Thanks @qibobo for the quick fix and @maximilien @zyjiaobj for the code review!
I'll merge it as it is blocking #22 Thanks!

**Pull request merged**